src/secrets/CommandRemove.ts

```diff
@@ -45,20 +47,19 @@ class CommandDelete extends CommandPolykey {
           nodeId: clientOptions.nodeId,
           host: clientOptions.clientHost,
           port: clientOptions.clientPort,
-          options: {
-            nodePath: options.nodePath,
-          },
+          options: { nodePath: options.nodePath },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsDelete({
+        await binUtils.retryAuthentication(async (auth) => {
+          for (const secretPath of secretPaths) {
+            await pkClient.rpcClient.methods.vaultsSecretsRemove({
               metadata: auth,
               nameOrId: secretPath[0],
-              secretName: secretPath[1],
-            }),
-          auth,
-        );
+              secretNames: [secretPath[1]],
+              options: { recursive: options.recursive },
+            });
+          }
+        }, meta);
```

Comment:
This isn't good. Removing all of the secrets needs to be done in a single `vaultsSecretRemove` RPC call. It needs to take the array of vault secret paths and do all the logic in the handler.

Alternatively for now we can just handle a single vault at a time. This probably makes more sense for now so all you need to do is check that all of the secret paths are for the same vault, if not then throw an error.

Reply:
I think I can sort the paths by the vault name and then split the groups based on that. Then, instead of sending one RPC call per file, we can do that per vault. Also, if we see that the vault name is undefined, we can try to look for the path on the system fs.

Reply:
In any case, minimise the processing on the client side. Have the RPC handler handle this logic.

Reply:
> In any case, minimise the processing on the client side. Have the RPC handler handle this logic.

Currently I am passing a single vault name as the `nameOrId` parameter of the RPC call. Instead of doing this, I can pass in the separated secret paths, so the handler can deal with the raw parsed input by itself. I don't like this because it differs from the parameter structure that every other command uses and would feel weird to use.

```ts
// Current implementation

const vaultName = 'vault';
const secretNames = ['secret1', 'secret2'];
await pkClient.rpcClient.methods.vaultsSecretsRemove({
  metadata: auth,
  nameOrId: vaultName,
  secretNames: secretNames,
  options: { recursive: options.recursive },
});

// This method

const secretNames = [
  ['vault1', 'secret1'], 
  ['vault2', 'secret2'],
];
await pkClient.rpcClient.methods.vaultsSecretsRemove({
  metadata: auth,
  secretNames: secretNames,
  options: { recursive: options.recursive },
});
```

Instead, I think I should do the calculation on client-side, grouping the secrets by the vault name. This would make the usage consistent for all RPC handlers.

```ts
// Current implementation
const secretNames = [
  ['vault1', 'secret1', 'secret2', 'secret3'], 
  ['vault2', 'secret4', 'secret5'],
];
for (const group of secretNames) {
  await pkClient.rpcClient.methods.vaultsSecretsRemove({
    metadata: auth,
    nameOrId: group.shift(), // Pop vault name
    secretNames: group, // Rest of the secrets
    options: { recursive: options.recursive },
  });
}
```
I like the second approach much more, but it comes at a cost of more processing for the client. I'm not sure which route to proceed with.

Reply:
I lean to keeping the logic more clustered to where it's needed. to me it makes the most sense to have the utility function `secretsRemove` deal with all the processing and grouping.

Reply:
Then I will proceed with the first method, passing a single `secretNames` instead of passing `nameOrId`. It will be expected that `secretNames` will have both the vault name and the secrets to delete from each vault.

Or maybe there is a better way to proceed with it?